zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/AuthorizationCheckBehavior.java

```diff
@@ -122,13 +95,50 @@ private Set<String> getAuthorizedResourceIdentifiers(
     return authorizationState.getResourceIdentifiers(userKey, resourceType, permissionType);
   }
 
-  private boolean hasWildcardPermission(final Set<String> authorizedResourceIdentifiers) {
-    return authorizedResourceIdentifiers.stream().anyMatch(WILDCARD_PERMISSION::equals);
-  }
-
   private boolean hasRequiredPermission(
       final Set<String> requiredResourceIdentifiers,
       final Set<String> authorizedResourceIdentifiers) {
     return authorizedResourceIdentifiers.stream().anyMatch(requiredResourceIdentifiers::contains);
   }
+
+  public static final class AuthorizationRequest<Resource> {
+    private final AuthorizationResourceType resourceType;
+    private final PermissionType permissionType;
+    private Resource resource;
+    private final Set<String> resourceIds;
+
+    public AuthorizationRequest(
+        final AuthorizationResourceType resourceType, final PermissionType permissionType) {
+      this.resourceType = resourceType;
+      this.permissionType = permissionType;
+      resourceIds = new HashSet<>();
+      resourceIds.add(WILDCARD_PERMISSION);
```

Comment:
👍 Adding this automatically makes a lot more sense, yes. We're always checking for this then when checking existing permissions 💡